### PR TITLE
refactor(filesystem): enhance prompt generation with dynamic templates 

### PR DESCRIPTION
**Pull request by JonXSnow**
Major Improvements:
- Implement dynamic system prompt generation using template-based approach
- Simplify to English-only support by removing i18n complexity
- Optimize config validation logic (reduce from 9 to 4 parameters)
- Add automatic fallback mechanism for error handling

Core Changes:
1. Dynamic Prompt Generation
   - Template-based prompt rendering with tool categorization
   - Auto-generate prompts based on available tools (file tools + execute tool)
   - Support custom tool names and disabled tools
   - Separate rendering for filesystem tools and execute tool

2. API Simplification
   - generateToolsPrompt: (string, error) → string (with internal error handling)
   - getToolDescription: Remove language parameter (English-only)
   - Add fallbackToStaticPrompt() for automatic error recovery

3. Data Structure Optimization
   - executeToolInfo: []toolInfo → *toolInfo (execute tool is always unique)
   - Simplify toolInfo struct (make it private)

4. Validation Enhancement
   - Simplify Config.Validate to check: nil config, nil backend, shell conflicts
   - Improve CustomTool field validation in ToolConfig

5. Tool Description Priority
   - Prioritize tool's own Desc field over fallback
   - Use getToolDescription() only when tool.Desc is empty
   - Better fallback messages for custom tools

Testing:
- Add comprehensive test suite with 34 test cases
- Test coverage: prompt generation, validation, custom tools, i18n
- Verify fallback behavior and edge cases

Code Quality:
- Add 467 lines of new functionality and tests
- Net change: 118 insertions, 110 deletions (improved code density)
- Reduce code duplication and improve maintainability

Breaking Changes:
- Removed Chinese prompt support (English-only)
- generateToolsPrompt no longer returns error
- getToolDescription requires no language parameter

This refactoring significantly improves the codebase by streamlining
the prompt generation logic, removing unused i18n complexity, and
providing a more maintainable architecture.

### CODE REVIEW

File: adk/middlewares/filesystem/prompt.go [hunk 1]
```diff
@@ -304,4 +314,195 @@ Use this tool to run commands, scripts, tests, builds, and other shell operation
...
+	filesystemToolsPromptTemplateCN = `# 文件系统工具{{range $i, $tool := .}}{{if $i}}、{{end}}'{{$tool.Name}}'{{end}}
+
+你可以使用这些工具与文件系统交互。
+所有文件路径必须以 '/' 开头。
+
+{{range .}}- {{.Name}}：{{.Description}}
+{{end}}`
+
+	// Template for generating dynamic execute tool prompt in English
+	executeToolPromptTemplateEN = `# Execute Tool {{range $i, $tool := .}}{{if $i}}, {{end}}'{{$tool.Name}}'{{end}}
```
Comment (hi-pender) [inline adk/middlewares/filesystem/prompt.go]: shell 工具，为啥要动态渲染。 这个就只有一个，要么有，要么没有

File: adk/middlewares/filesystem/filesystem.go [hunk 2]
```diff
@@ -294,23 +300,30 @@ func (c *MiddlewareConfig) Validate() error {
...
+		if cfg != nil && cfg.CustomTool != nil {
+			hasCustomTool = true
+			break
+		}
+	}
+
+	if !hasBackendOrShell && !hasCustomTool {
+		return errors.New("at least one of Backend, Shell, StreamingShell, or custom tool must be set")
 	}
 	if streamingShell != nil && shell != nil {
```
Comment (hi-pender) [inline adk/middlewares/filesystem/filesystem.go]: 这个语义也不对。 hasCustomToolConfig 和 hasCustomTool 是两码事儿

Comment (hi-pender) [reply adk/middlewares/filesystem/filesystem.go]: 这里最终想确实的是：是否最终有 Tool。

Comment (hi-pender) [reply adk/middlewares/filesystem/filesystem.go]: 简化一点的设定：
1. Backend 必须存在。
2. Shell、StreamingShell 可选存在。

File: adk/middlewares/filesystem/prompt.go [hunk 3]
```diff
@@ -296,12 +291,114 @@ Use this tool to run commands, scripts, tests, builds, and other shell operation
...
+type toolInfo struct {
+	Name        string
+	Description string
+}
+
+// getToolDescription returns the English description for a given tool name.
+// It provides short descriptions used in system prompts.
+func getToolDescription(toolName string) string {
+	descriptions := map[string]string{
+		ToolNameLs:        "list files in a directory (requires absolute path)",
```
Comment (hi-pender) [inline adk/middlewares/filesystem/prompt.go:324]: 如果别人指定了 ToolName，这里的描述全都变成「filesystem tool, refer to the tool's description for details」
